File: app/src/main/java/mrkm4ntr/twitterclient/data/TwitterDbHelper.java
package mrkm4ntr.twitterclient.data;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class TwitterDbHelper extends SQLiteOpenHelper {

    private static final int DATABASE_VERSION = 1;
    private static final String DATABASE_NAME = "twitter.db";

    public TwitterDbHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    private static final String SQL_CREATE_STATUS_TABLE =
            String.format("CREATE TABLE %s (%s INTEGER PRIMARY KEY ON CONFLICT REPLACE, %s TEXT NOT NULL, %s INTEGER NOT NULL, %s TEXT NOT NULL, %s TEXT NOT NULL)",
                    TwitterContract.StatusEntry.TABLE_NAME, TwitterContract.StatusEntry._ID,
                    TwitterContract.StatusEntry.COLUMN_CREATE_AT,
                    TwitterContract.StatusEntry.COLUMN_TEXT,
                    TwitterContract.StatusEntry.COLUMN_USER_NAME,
                    TwitterContract.StatusEntry.COLUMN_USER_PROFILE_IMAGE_URL);

    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase) {
        sqLiteDatabase.execSQL(SQL_CREATE_STATUS_TABLE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS " + TwitterContract.StatusEntry.TABLE_NAME);
        onCreate(sqLiteDatabase);
    }
}

File: app/src/main/kotlin/mrkm4ntr/twitterclient/data/TwitterDbHelper.kt
package mrkm4ntr.twitterclient.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class TwitterDbHelper(context: Context) : SQLiteOpenHelper(
        context, TwitterDbHelper.DATABASE_NAME, null, TwitterDbHelper.DATABASE_VERSION) {

    override fun onCreate(sqLiteDatabase: SQLiteDatabase) {
        sqLiteDatabase.execSQL(SQL_CREATE_STATUS_TABLE)
        sqLiteDatabase.execSQL(SQL_CREATE_ACCOUNT_TABLE)
    }

    override fun onUpgrade(sqLiteDatabase: SQLiteDatabase, i: Int, i1: Int) {
        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS ${TwitterContract.StatusEntry.TABLE_NAME}")
        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS ${TwitterContract.AccountEntry.TABLE_NAME}")
        onCreate(sqLiteDatabase)
    }

    companion object {

        private val DATABASE_VERSION = 1
        private val DATABASE_NAME = "twitter.db"

        private val SQL_CREATE_STATUS_TABLE = """
            CREATE TABLE ${TwitterContract.StatusEntry.TABLE_NAME} (
                ${TwitterContract.StatusEntry._ID} INTEGER PRIMARY KEY ON CONFLICT REPLACE,
                ${TwitterContract.StatusEntry.COLUMN_CREATE_AT} TEXT NOT NULL,
                ${TwitterContract.StatusEntry.COLUMN_TEXT} INTEGER NOT NULL,
                ${TwitterContract.StatusEntry.COLUMN_USER_NAME} TEXT NOT NULL,
                ${TwitterContract.StatusEntry.COLUMN_USER_PROFILE_IMAGE_URL} TEXT NOT NULL,
                ${TwitterContract.StatusEntry.COLUMN_USER_SCREEN_NAME} TEXT NOT NULL,
                ${TwitterContract.StatusEntry.COLUMN_USER_LOCATION} TEXT NOT NULL,
                ${TwitterContract.StatusEntry.COLUMN_USER_BIO} TEXT NOT NULL
            )"""

        private val SQL_CREATE_ACCOUNT_TABLE = """
            CREATE TABLE ${TwitterContract.AccountEntry.TABLE_NAME} (
                ${TwitterContract.AccountEntry._ID} INTEGER PRIMARY KEY ON CONFLICT REPLACE,
                ${TwitterContract.AccountEntry.COLUMN_NAME} TEXT NOT NULL,
                ${TwitterContract.AccountEntry.COLUMN_SCREEN_NAME} TEXT NOT NULL,
                ${TwitterContract.AccountEntry.COLUMN_PROFILE_IMAGE_URL} TEXT NOT NULL,
                ${TwitterContract.AccountEntry.COLUMN_PROFILE_BANNER_URL} TEXT NOT NULL,
                UNIQUE (${TwitterContract.AccountEntry.COLUMN_SCREEN_NAME}) ON CONFLICT REPLACE
            )"""
    }
}
